### DISCUSSION TITLE
Retry strategy after ECONNRESET?

### DISCUSSION THREAD

**Discussion by ilg-ul** [node-fetch/node-fetch#1708] [Q&A]
I am occasionally getting an `read ECONNRESET` while reading via a pipeline, during GitHub actions that need to download binary archives with tools, to satisfy dependencies (via `xpm` & xPacks).

Currently I'm manually restarting the failed jobs, but I was thinking perhaps implementing a small loop in `xpm`, with several retries, might fix these issues.

Did anyone try such solutions? What would be the best strategy? Is there any way to reuse the connection, or a new `fetch()` is required? Is there a recommended delay between retries, to avoid upsetting the GitHub server with too many requests?

**Comment by dude0001** [comment]
@ilg-ul did you end up find a solution for this?
